gateway/ln-gateway/src/gateway.rs

```diff
@@ -44,29 +50,43 @@ impl Gateway {
     pub async fn new(
         config: GatewayConfig,
         decoders: ModuleDecoderRegistry,
-        ln_rpc: Arc<dyn LnRpc>,
+        lnrpc_factory: DynLnRpcClientFactory,
         client_builder: DynGatewayClientBuilder,
-        // TODO: consider encapsulating message channel within LnGateway
-        sender: mpsc::Sender<GatewayRequest>,
-        receiver: mpsc::Receiver<GatewayRequest>,
         task_group: TaskGroup,
     ) -> Self {
+        // Create message channels for the webserver
+        let (sender, receiver) = mpsc::channel::<GatewayRequest>(100);
+
+        // Start gateway without an lnrpc client
+        let lnrpc: Option<DynLnRpcClient> = None;
```

Comment:
Should be initialized here imo.

Also, if there is only one LN rpc client, then we actually don't need the factory. Allowing hot swaps of the underlying LN node is not a current goal imo. That would require a whole lot of complicated shutdown logic that we don't have and we should keep it as simple as possible till we actually want that feature.